### PR TITLE
Add startpage search engine

### PR DESCRIPTION
**Pull request by clementpoiret**
Dear @deepjyoti30, thanks for your awesome extension.

I'd like to add an additional search engine, startpage, to have proxied google results with more privacy :)

### CODE REVIEW

**Comment by clementpoiret** [comment]
what about adding a "bang" notation as in duckduckgo?
Like: my main search engine is Startpage, but I want to search on stackoverflow this time so I can write for example: `!s dumb question on stackoverflow"

**Comment by reesericci** [comment]
I think that would be awesome!

**Comment by deepjyoti30** [comment]
@clementpoiret

>Like: my main search engine is Startpage, but I want to search on stackoverflow this time so I can write for example: `!s dumb question on stackoverflow"

Isn't the quicklink feature the same thing here? You can define `s` to map to `https://stackoverflow.com` and you can type `s'dumb question on stackoverflow` to search for the text on stackoverflow.

File: js/main.js [hunk 1]
```diff
@@ -37,7 +37,13 @@ searchEngines = {
...
     "Yahoo": "https://search.yahoo.com/search?p=",
     "Ecosia": "https://www.ecosia.org/search?q=",
-    "Startpage": "https://startpage.com/do/search?query="
+    "Startpage": "https://startpage.com/do/search?query=",
+    "Yandex": "https://yandex.com/search/?text=",
+    "Qwant": "https://www.qwant.com/?q=",
+    "Wikipedia": "https://wikipedia.org/wiki/Special:Search?search=",
+    "Wolfram": "https://www.wolframalpha.com/input/?i=",
+    "Baidu": "https://www.baidu.com/s?wd=",
+    "StackOverflow": "https://stackoverflow.com/search?q="
```
Comment (deepjyoti30) [inline js/main.js:46]: @clementpoiret I am cool with defining more generic search engines here, like `startpage` and `google`. 

But I think `stackoverflow`, `wikipedia` are not generic search engines and should instead be defined using the quicklink feature.

Could you just keep the generic search engines in this PR?

I would say following are cool to keep:
- `startpage`
- `baidu`
- `yandex`
- `qwant`

Others can be defined by the user using the [quicklink feature that was added in v0.4](https://github.com/deepjyoti30/startpage/wiki/Configuration#customizing-the-quicklinks)

**Review by deepjyoti30 — CHANGES_REQUESTED** [review]

File: js/main.js (hunk 1 above)
Comment (clementpoiret) [reply js/main.js:46]: I agree with you.
I also think the bang notation is equivalent to quickLinks, so are you okay with defining search engines as quickLinks by default too? As many user are using engines like qwant or ddg for general searches, but are switching to google or startpage for specific queries?

Comment (deepjyoti30) [reply js/main.js]: Yes, I am cool with predefining them through the config file.

We are currently defining `GitHub`, `YouTube` and `Reddit` but we can add more for the users convenience. Let's make it a separate PR if you want to work on it!

**Pull request merged**